test/python/python_request_response_reliability_service.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Provizio Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import sys
+import threading
+import time
+
+import provizio_dds
+
+_START_TIME = time.monotonic()
+
+
+def _timestamp() -> str:
+    seconds = time.monotonic() - _START_TIME
+    return f"[{seconds:0.3f}] "
+
+
+def main() -> int:
+    if len(sys.argv) != 4:
+        print(
+            "Usage: python_request_response_reliability_service.py "
+            "<test_name_postfix> <first_iteration_value> <num_iterations>"
+        )
+        return 1
+
+    test_name_postfix = sys.argv[1]
+    expected_value = int(sys.argv[2])
+    num_iterations = int(sys.argv[3])
+    initial_iterations = max(1, num_iterations)
+    wait_timeout = num_iterations * 3.0 + 5
+
+    log_prefix = f"python_request_response_reliability_service{test_name_postfix}: "
+    service_name = (
+        f"provizio_dds_test_request_response_reliability{test_name_postfix}"
+    )
+    domain_id = 14
+
+    random.seed(expected_value)
+    wait_ms = random.randint(0, 1999)
+    if wait_ms >= 1000:
+        print(f"{log_prefix}{_timestamp()}Waiting {wait_ms}ms...")
```

Comment:
This can be done automatically with a proper logger, instead of having to define the log_prefix.

Reply:
Sure, but same comment - it's a unit test, my intention was to keep it as simple as it gets and only test the functionality the test intends to

Reply:
All good 👍